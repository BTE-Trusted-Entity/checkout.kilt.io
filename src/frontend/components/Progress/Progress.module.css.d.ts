export const __esModule: true;
export const container: string;
export const step: string;
export const current: string;
export const completed: string;
