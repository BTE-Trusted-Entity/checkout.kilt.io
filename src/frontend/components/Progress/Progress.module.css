.container {
  width: 100%;
  list-style: none;
  display: flex;
  padding: 0;
  border: 1px solid var(--color-progress);

  --color-progress: #3a0029;
}

.step {
  flex-grow: 1;
  border: 1px solid var(--color-progress);
  padding: 0.25rem 0.75rem;
  text-transform: uppercase;
  font-size: 0.75rem;
  line-height: 0.75rem;
  color: rgba(var(--color-white-rgb) / 50%);
}

.current {
  composes: step;
  color: var(--color-white);

  --color-transparent: rgba(var(--color-white-rgb) / 0%);

  background-image:
    repeating-linear-gradient(
      120deg,
      var(--color-transparent),
      var(--color-transparent) 5px,
      var(--color-primary) 5px,
      var(--color-primary) 10px
    );
}

.completed {
  composes: step;
  color: var(--color-white);
  background: var(--color-progress);
}
