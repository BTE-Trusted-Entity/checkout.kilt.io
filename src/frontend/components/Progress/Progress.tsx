import * as styles from './Progress.module.css';

export function Progress({
  stages,
  current,
}: {
  stages: string[];
  current: string;
}) {
  const currentIndex = stages.indexOf(current);

  function getClass(index: number) {
    if (index < currentIndex) return styles.completed;
    if (index === currentIndex) return styles.current;
    return styles.step;
  }

  return (
    <ul className={styles.container}>
      {stages.map((stage, index) => (
        <li key={stage} className={getClass(index)}>
          {stage}
        </li>
      ))}
    </ul>
  );
}
