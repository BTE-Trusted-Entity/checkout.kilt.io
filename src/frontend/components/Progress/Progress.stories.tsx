import { Meta } from '@storybook/react';

import { Progress } from './Progress';

export default {
  title: 'Components/Progress',
  component: Progress,
} as Meta;

export function Template(): JSX.Element {
  return (
    <Progress stages={['Prepare', 'Order', 'Pay', 'Success']} current="Order" />
  );
}
