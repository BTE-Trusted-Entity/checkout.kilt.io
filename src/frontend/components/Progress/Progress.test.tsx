import { describe, it } from '@jest/globals';

import { render } from '../../../testing/testing';

import { Progress } from './Progress';

describe('Progress', () => {
  it('should match the snapshot', async () => {
    const { container } = render(
      <Progress
        stages={['Prepare', 'Order', 'Pay', 'Success']}
        current="Order"
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
