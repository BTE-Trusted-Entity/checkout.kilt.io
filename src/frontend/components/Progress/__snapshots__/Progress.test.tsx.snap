// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress should match the snapshot 1`] = `
<div>
  <ul
    class="container"
  >
    <li
      class="completed"
    >
      Prepare
    </li>
    <li
      class="current"
    >
      Order
    </li>
    <li
      class="step"
    >
      Pay
    </li>
    <li
      class="step"
    >
      Success
    </li>
  </ul>
</div>
`;
